geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/databasehandlers/MbtilesDatabaseHandler.java

```diff
@@ -152,18 +152,61 @@ public boolean isValid() {
     public List<SpatialVectorTable> getSpatialVectorTables( boolean forceRead ) throws Exception {
         return Collections.emptyList();
     }
-
+    
+    /**
+     * Load list of Table [Raster] for MbTiles using  [SpatialRasterTable]
+     * 
+     * <p>[GeneralQueriesPreparer] Documentation of vector_key /  vector_value   
+     * <ol>
+     * <li>vector_key: 6 Fields will be returned with the following structure[They may not be empty, otherwise lenght of split will not return the correct amount]</li>
+     * <li>0 table_name: databaseFileNameNoExtensiony/li>
+     * <li>1: geometry_column - not used '0'</li>
+     * <li>2: layer_type - SpatialDataType.MBTILES.getTypeName()</li>
+     * <li>3: ROWID - short description[databaseFileNameNoExtension]</li>
+     * <li>4: view_read_only - long description [tableName]</li>
+     * </ol>
+     * <li>vector_data: Seperator: ';' 7 values minimum [more must be made known to parse_vector_key_value()]</li>
+     * <li>0: geometry_type - this.minZoom</li>
+     * <li>1: coord_dimension - this.maxZoom</li>
+     * <li>2: srid - 4326</li>
+     * <li>3: spatial_index_enabled - not used '0'</li>
+     * <li>4: rows - not used '0'
+     * <li>5: extent_min/max - Seperator ',' - 4 values [mbtiles,map,mapurl: 7 values : centerX,centerY,defaultZoom]
+     * <li>5.1:extent_min_x - boundsWest</li>
+     * <li>5.2:extent_min_y - boundsSouth</li>
+     * <li>5.3:extent_max_x - boundsEast</li>
+     * <li>5.4:extent_max_y - boundsNorth</li>
+     * <li>5.5: centerX - can be user defined</li>
+     * <li>5.6: centerY - can be user defined</li>
+     * <li>5.7: defaultZoom - can be user defined</li>
+     * <li>6:last_verified - tileQuery as '?,?,?'</li>
+     * <li>7:getDatabasePath()</li>
+     * <li>8-?:not used</li>
+     * </ol>
+     * <p>vector_key and vector_value
+     * <ol>
+     * <li>vector_key[mbtiles]='databaseFileNameNoExtension;0;MBTiles;title;abstract' = length=5[0-4]</li>
+     * <li>vector_data[mbtiles]='minZoom;maxZoom;srid;0' = length=4[0-3]</li>
+     * <li>vector_extent[mbtiles]='0;extent_0-6;?,?,?;getDatabasePath()' = length=4[0-3]</li>
+     * </ol>
+     * Note: title and abstract (short/long descriptions) replace ';' with '-'
+     * <p> vector_key/data documetation : 20150718
+     * 
+     * @return list of RasterTabes.
+     */     
     public List<SpatialRasterTable> getSpatialRasterTables( boolean forceRead ) throws Exception {
```

Comment:
2015-09-19 14:23 GMT+02:00 Andrea Antonello <notifications@github.com>:

> In
> geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/databasehandlers/MbtilesDatabaseHandler.java
> <https://github.com/geopaparazzi/geopaparazzi/pull/287#discussion_r39919113>
> :
>
> > +     * <li>5.7: defaultZoom - can be user defined</li>
> > +     * <li>6:last_verified - tileQuery as '?,?,?'</li>
> > +     * <li>7:getDatabasePath()</li>
> > +     * <li>8-?:not used</li>
> > +     * </ol>
> > +     * <p>vector_key and vector_value
> > +     * <ol>
> > +     * <li>vector_key[mbtiles]='databaseFileNameNoExtension;0;MBTiles;title;abstract' = length=5[0-4]</li>
> > +     * <li>vector_data[mbtiles]='minZoom;maxZoom;srid;0' = length=4[0-3]</li>
> > +     * <li>vector_extent[mbtiles]='0;extent_0-6;?,?,?;getDatabasePath()' = length=4[0-3]</li>
> > +     * </ol>
> > +     * Note: title and abstract (short/long descriptions) replace ';' with '-'
> > +     * <p> vector_key/data documetation : 20150718
> > +     *
> > +     * @return list of RasterTabes.
> > +     */
> >      public List<SpatialRasterTable> getSpatialRasterTables( boolean forceRead ) throws Exception {
>
> From here I can't see it. Is this method available from an interface or
> abstract class? If yes, the javadoc should be placed only there. I have the
> feeling I have seen that javadoc a few times already. Is it the same and
> can be centralized for better maintainance?
>

The main documentation is in
* AbstractSpatialTable.parse_vector_key_value
   * the only place they are now being **read** / parsed

They are **created** in different places (above mbtiles)
* there I added it also to show how they were used

> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geopaparazzi/geopaparazzi/pull/287/files#r39919113>.
>